common-libs/common/src/services/DidDocumentService/LocalKeyVault.ts

```diff
@@ -0,0 +1,46 @@
+import { KeyVault } from './KeyVault'
+import { KeysService } from '../index'
+import * as secp256k1 from 'secp256k1'
+
+/**
+ * KeyStore backed implementation when the seed is known
+ */
+export class LocalKeyVault implements KeyVault {
```

Comment:
Will there be any other keyvaults?

Reply:
Yes that's the plan, e.g in the web3 experiment we'll use metamask as the provider of keys.

Reply:
Also what is the point of making it a class instead of a plain object, if you have declared `KeyVault` interface anyway?

```
export const createKeyVault = (keysService: KeysService) => ({
  getPrimaryPublicKey: () => ...,
  ...
})
}

Reply:
Ease of use, It's a common use case to use the local seed. In this case all a client needs to do is call `new LocalKeyVault`

Reply:
They could call `createKeyVault` all the same

Reply:
Back to coding style, I'm happy to use any approach the team agrees on.

Reply:
For now this is working let's update it when needed.